### PR TITLE
generator-electrode: [patch] [chore] Set ES6 flag in gulpfile.

### PR DESCRIPTION
**Pull request by animesh10**
- set ES6 test flag that is used by the `arch-gulpfile`.
- `test/server` eslint file extends from the new es6 server test file.

### CODE REVIEW

File: packages/generator-electrode/generators/app/templates/test/server/eslintrc [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
```
Comment (jchip) [inline packages/generator-electrode/generators/app/templates/test/server/eslintrc]: where is this file being copy in the code?

File: packages/generator-electrode/generators/app/templates/gulpfile.js [hunk 2]
```diff
@@ -1 +1,2 @@
+process.env.ES6_LINT = true;
```
Comment (jchip) [inline packages/generator-electrode/generators/app/templates/gulpfile.js]: `process.env.SERVER_ES6`

Doesn't the archetype need to look at this flag and decide to use ES6 eslint to lint the code in `src/server`?

File: packages/generator-electrode/generators/app/templates/test/server/eslintrc (hunk 1 above)
Comment (animesh10) [inline packages/generator-electrode/generators/app/templates/test/server/eslintrc]: the entire test folder is being copied [here](https://github.com/electrode-io/electrode/blob/master/packages/generator-electrode/generators/app/index.js#L329). Then eslint files moved [here](https://github.com/electrode-io/electrode/blob/master/packages/generator-electrode/generators/app/index.js#L378)

File: packages/generator-electrode/generators/app/templates/gulpfile.js (hunk 2 above)
Comment (animesh10) [inline packages/generator-electrode/generators/app/templates/gulpfile.js]: The `lint-server` task is using  [.eslintrc-node](https://github.com/electrode-io/electrode/blob/master/packages/electrode-archetype-react-app-dev/config/eslint/.eslintrc-node) which in turn is extending [es6-node](https://github.com/electrode-io/electrode/blob/master/packages/electrode-archetype-react-app-dev/config/eslint/.eslintrc-node#L3)

Comment (jchip) [inline packages/generator-electrode/generators/app/templates/gulpfile.js]: then what's the point of setting this?

Comment (animesh10) [inline packages/generator-electrode/generators/app/templates/gulpfile.js]: The task `lint-server-test` is using `.eslintrc-mocha-test` which is extending `es5-node` configurations. 
This flag will switch the new generated apps to use the `eslintrc-mocha-test-es6` config which extends `es6-node` config for the `lint-server-test` task.

**Pull request merged**